File: app/src/main/java/com/supercharge/Homework/model/MovieBO.java
package com.supercharge.Homework.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class MovieBO {
    @SerializedName("id")
    @Expose
    private int id;

    @SerializedName("title")
    @Expose
    private String title;

    @SerializedName("overview")
    @Expose
    private String overview;

    @SerializedName("popularity")
    @Expose
    private double popularity;

    @SerializedName("backdrop_path")
    @Expose
    private String posterPath;

    @SerializedName("poster_path")
    @Expose
    private String picture;

    @SerializedName("release_date")
    @Expose
    private String releaseDate;

    @SerializedName("vote_average")
    @Expose
    private double vote_average;

    @SerializedName("genre_ids")
    @Expose
    private List<Integer> genreIds;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getOverview() {
        return overview;
    }

    public void setOverview(String overview) {
        this.overview = overview;
    }

    public double getPopularity() {
        return popularity;
    }

    public void setPopularity(double popularity) {
        this.popularity = popularity;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public void setPosterPath(String posterPath) {
        this.posterPath = posterPath;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(String releaseDate) {
        this.releaseDate = releaseDate;
    }

    public double getVote_average() {
        return vote_average;
    }

    public void setVote_average(double vote_average) {
        this.vote_average = vote_average;
    }

    public List<Integer> getGenreIds() {
        return genreIds;
    }

    public void setGenreIds(List<Integer> genreIds) {
        this.genreIds = genreIds;
    }

    public String getPicture() {
        return picture;
    }

    public void setPicture(String picture) {
        this.picture = picture;
    }
}

File: app/src/main/java/com/supercharge/Homework/model/MovieBO.kt
package com.supercharge.Homework.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class MovieBO {
    @SerializedName("id")
    @Expose
    val id: Int = 0

    @SerializedName("title")
    @Expose
    val title: String? = null

    @SerializedName("overview")
    @Expose
    val overview: String? = null

    @SerializedName("popularity")
    @Expose
    val popularity: Double = 0.toDouble()

    @SerializedName("backdrop_path")
    @Expose
    val posterPath: String? = null

    @SerializedName("poster_path")
    @Expose
    val picture: String? = null

    @SerializedName("release_date")
    @Expose
    val releaseDate: String? = null

    @SerializedName("vote_average")
    @Expose
    val vote_average: Double = 0.toDouble()

    @SerializedName("genre_ids")
    @Expose
    val genreIds: List<Int>? = null

}